doc/user/shared-content/namespaces-content.md

```diff
@@ -0,0 +1,120 @@
+Namespaces are a way to organize Materialize objects logically. In organizations
+with multiple objects, namespaces help avoid naming conflicts and make it easier
+to manage objects.
+
+## Namespace hierarchy
+
+Materialize follows SQL standard's namespace hierarchy for most objects (for the
+exceptions, see [Other objects](#other-objects)).
+
+|                           |             |
+|---------------------------| ------------|
+| 1st/Highest level:        |  **Database** |
+| 2nd level:                |  **Schema**   |
+| 3rd level:                | <table><tbody><tr><td><ul><li>**Table**</li><li>**View**</li><li>**Materialized view**</li><li>**Connection**</li></ul></td><td><ul><li>**Source**</li><li>**Sink**</li><li>**Index**</li></ul></td><td><ul><li>**Type**</li><li>**Function**</li><li>**Secret**</li></ul></td></tr></tbody></table>|
+| Lowest level:             | **Column**     |
+
+Each layer in the hierarchy can contain elements from the level immediately
+beneath it. That is,
+
+- Databases can contain: schemas;
+- Schemas can contain: tables, views, materialized views, connections, sources,
+sinks, indexes, types, functions, and secrets;
+- Tables, views, and materialized views can contain: columns.
+
+
+### Qualifying names
+
+Namespaces enable disambiguation and access to objects across different
+databases and schemas. Namespaces use the dot notation format
+(`<database>.<schema>....`) and allow you to refer to objects by:
+
+- **Fully qualified names**
+
+  Used to reference objects in a different database (Materialize allows
+  cross-database queries); e.g.,
+
+  ```
+  <Database>.<Schema>
+  <Database>.<Schema>.<Source>
+  <Database>.<Schema>.<View>
+  <Database>.<Schema>.<Table>.<Column>
+  ```
+
+  {{< tip >}}
+
+  You can use fully qualified names to reference objects within the same
+  database (or within the same database and schema). However, for brevity and
+  readability, you may prefer to use qualified names instead.
+
+  {{</ tip >}}
+
+- **Qualified names**
+
+  - Used to reference objects within the same database but different schema, use
+    the schema and object name; e.g.,
+
+    ```
+    <Schema>.<Source>
+    <Schema>.<View>
+    <Schema>.<Table>.<Column>
+    ```
+
+  - Used to reference objects within the same database and schema, use the
+    object name; e.g.,
+
+    ```
+    <Source>
+    <View>
+    <Table>.<Column>
+    <View>.<Column>
+    ```
+
+## Namespace constraints
+
+- All namespaces must adhere to [identifier rules](/sql/identifiers).
```

Comment:
nit: only use bullet points if there are multiple options? but understand if this is there to be consistent between sub headings? or be clear that there is a fact here?
```suggestion
All namespaces must adhere to [identifier rules](/sql/identifiers).
```